templates/vue-demo-store/composables/useCart.ts

```diff
@@ -11,6 +11,7 @@ const _useCart = (): UseCartReturn => {
   ): Promise<Schemas["Cart"]> {
     if (newCart) {
       _storeCart.value = newCart;
+      useCartData.refreshCart(_storeCart.value);
```

Comment:
There is a fix for that https://github.com/shopware/frontends/pull/506/files

Maintaining 2 files with the same code is not a good idea. That is why I make it more of a proxy in my MR. Also, it is fixing the doubled state problem.